example_workflow_mvp/workflow_mvp.py

```diff
@@ -0,0 +1,166 @@
+# This logging is needed for debuggin class initalization.
+# import logging
+
+# log_format = "%(levelname).1s%(asctime)s %(filename)s:%(lineno)d] %(message)s"
+# date_format = "%m%d %H:%M:%S"
+# logging.basicConfig(level=logging.DEBUG, format=log_format, datefmt=date_format)
+
+
+import random
+import string
+from typing import Protocol
+
+import pydantic
+import slay
+from user_package import shared_processor
+
+IMAGE_COMMON = slay.Image().pip_requirements_txt("common_requirements.txt")
+
+
+class GenerateData(slay.ProcessorBase):
+
+    default_config = slay.Config(image=IMAGE_COMMON)
+
+    def run(self, length: int) -> str:
+        return "".join(random.choices(string.ascii_letters + string.digits, k=length))
+
+
+IMAGE_TRANSFORMERS_GPU = (
+    slay.Image()
+    .cuda("12.8")
+    .pip_requirements_txt("common_requirements.txt")
+    .pip_install("transformers")
+)
+
+
+class MistraLLMConfig(pydantic.BaseModel):
+    hf_model_name: str
+
+
+class MistralLLM(slay.ProcessorBase[MistraLLMConfig]):
+
+    default_config = slay.Config(
+        image=IMAGE_TRANSFORMERS_GPU,
+        resources=slay.Resources().cpu(12).gpu("A100"),
+        user_config=MistraLLMConfig(hf_model_name="EleutherAI/mistral-6.7B"),
+    )
+    # default_config = slay.Config(config_path="mistral_config.yaml")
+
+    def __init__(
+        self,
+        context: slay.Context = slay.provide_context(),
+    ) -> None:
+        super().__init__(context)
+        # import subprocess
+        # try:
+        #     subprocess.check_output(["nvidia-smi"], text=True)
+        # except:
+        #     raise RuntimeError(
+        #         f"Cannot run `{self.__class__}`, because host has no CUDA."
+        #     )
+        # import transformers
+
+        # model_name = self.user_config.hf_model_name
+        # tokenizer = transformers.AutoTokenizer.from_pretrained(model_name)
+        # model = transformers.AutoModelForCausalLM.from_pretrained(model_name)
+        # self._model = transformers.pipeline(
+        #     "text-generation", model=model, tokenizer=tokenizer
+        # )
+
+    def run(self, data: str) -> str:
+        return data.upper()
+        # return self._model(data, max_length=50)
+
+
+class MistralP(Protocol):
+    def __init__(self, context: slay.Context) -> None:
+        ...
+
+    def run(self, data: str) -> str:
+        ...
+
+
+class TextToNum(slay.ProcessorBase):
+    default_config = slay.Config(image=IMAGE_COMMON)
+
+    def __init__(
+        self,
+        context: slay.Context = slay.provide_context(),
+        mistral: MistralP = slay.provide(MistralLLM),
+    ) -> None:
+        super().__init__(context)
+        self._mistral = mistral
+
+    def run(self, data: str) -> int:
+        number = 0
+        generated_text = self._mistral.run(data)
+        for char in generated_text:
+            number += ord(char)
+
+        return number
+
+
+class Workflow(slay.ProcessorBase):
+    default_config = slay.Config(image=IMAGE_COMMON)
```

Comment:
could we have an example that still uses the file-based config (if we still want to support that)

Reply:
Yes - though the whole config topic seems to need some more thought overall. I'm currently just drafting something here for demo purposes.

https://linear.app/baseten/issue/BT-10091/support-hybrid-filecode-based-config